### PR TITLE
feat: add C implementation for `math/base/special/lcm`

### PR DESCRIPTION
**Pull request by gunjjoshi**
## Description

> What is the purpose of this pull request?

This pull request:

-   adds C implementation for [`math/base/special/lcm`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/math/base/special/lcm).

```c
double stdlib_base_lcm( const double a, const double b )
```

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves a part of #649.

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

**Review by Planeshifter — APPROVED** [review]
LGTM. Thanks, @gunjjoshi!

File: lib/node_modules/@stdlib/math/base/special/lcm/src/main.c [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+* @example
+* double out = stdlib_base_lcm( 21.0, 6.0 );
+* // returns 42.0
+*/
+double stdlib_base_lcm( const double a, const double b ) {
+	double an;
+	double bn;
+	double d;
+
+	if( a == 0.0 || b == 0.0 ) {
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/lcm/src/main.c:40]: Missing space.

File: lib/node_modules/@stdlib/math/base/special/lcm/src/main.c [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+		return 0.0;
+	}
+	if ( a < 0.0 ) {
+		an = -a;
+	}
+	else {
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/lcm/src/main.c:46]: No need for line break.

File: lib/node_modules/@stdlib/math/base/special/lcm/src/main.c [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+		an = a;
+	}
+	if ( b < 0.0 ) {
+		bn = -b;
+	}
+	else {
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/lcm/src/main.c:52]: No need for linebreak.

File: lib/node_modules/@stdlib/math/base/special/lcm/src/main.c [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+*/
+
+#include "stdlib/math/base/special/lcm.h"
+#include "stdlib/math/base/assert/is_nan.h"
+#include "stdlib/math/base/special/gcd.h"
+
+/**
+* Computes the least common multiple (lcm).
+*
+* @private
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/lcm/src/main.c:26]: This isn't a private API. The `@private` tag should be removed.

**Comment by kgryte** [comment]
@gunjjoshi Would you mind submitting a small follow-up PR addressing the comments above? Thanks!

**Comment by gunjjoshi** [comment]
> @gunjjoshi Would you mind submitting a small follow-up PR addressing the comments above? Thanks!

Sure, I've made a PR #2296.

**Pull request merged**